test/dlc_tests/sanity/test_dlc_labels.py

```diff
@@ -21,35 +19,91 @@ def test_dlc_major_version_label(image, region):
     :param region: <str> region where ECR repository holding the image resides
     :return:
     """
-    ecr_client = boto3.client("ecr", region_name=region)
-
-    image_repository, image_tag = test_utils.get_repository_and_tag_from_image_uri(image)
-    # Using "acceptedMediaTypes" on the batch_get_image request allows the returned image information to
-    # provide the ECR Image Manifest in the specific format that we need, so that the image LABELS can be found
-    # on the manifest. The default format does not return the image LABELs.
-    response = ecr_client.batch_get_image(
-        repositoryName=image_repository,
-        imageIds=[{"imageTag": image_tag}],
-        acceptedMediaTypes=["application/vnd.docker.distribution.manifest.v1+json"],
-    )
-    if not response.get("images"):
-        raise KeyError(
-            f"Failed to get images through ecr_client.batch_get_image response for image {image_repository}:{image_tag}"
-        )
-    elif not response["images"][0].get("imageManifest"):
-        raise KeyError(f"imageManifest not found in ecr_client.batch_get_image response:\n{response['images']}")
-
-    manifest_str = response["images"][0]["imageManifest"]
-    # manifest_str is a json-format string
-    manifest = json.loads(manifest_str)
-    image_metadata = json.loads(manifest["history"][0]["v1Compatibility"])
-    major_version = image_metadata["config"]["Labels"].get("dlc_major_version", None)
+    labels = test_utils.get_labels_from_ecr_image(image, region)
+    major_version = labels.get("dlc_major_version", None)
 
     assert major_version, f"{image} has no LABEL named 'dlc_major_version'. Please insert label."
 
     test_utils.LOGGER.info(f"{image} has 'dlc_major_version' = {major_version}")
 
 
+@pytest.mark.usefixtures("sagemaker")
+@pytest.mark.integration("dlc_labels")
+@pytest.mark.model("N/A")
+def test_dlc_standard_labels(image, region):
```

Comment:
Just to ensure, this test would not run on the canaries, right?

Reply:
these don't run on canaries but can be added by using the canary marker. Though they will fail for images that don't have the labels, so if added as canaries, we should ensure we have added labels to all of the images

Reply:
Yeah exactly, I wanted to confirm that we do not run them on canaries. Thanks!!